rules/aws_s3_bucket_name.go

```diff
@@ -51,6 +64,68 @@ func (r *AwsS3BucketNameRule) Link() string {
 	return project.ReferenceLink(r.Name())
 }
 
+// Return regex bucket naming rules
+func rules() ([]map[string]string, error) {
+	// Get bucket naming rule based on length from AWS SDK API models
+	r, err := http.Get(
+		"https://raw.githubusercontent.com/aws/aws-sdk-go/main/models/apis/cloudformation/2010-05-15/api-2.json",
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer r.Body.Close()
+
+	bucket := Bucket{}
+	err = json.NewDecoder(r.Body).Decode(&bucket)
+	if err != nil {
+		return nil, err
+	}
+
+	// https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html
+	return []map[string]string{
+		{
+			"regx": fmt.Sprintf(
+				"^(.{0,%d})?(.{%d,})?$",
+				bucket.Shapes.S3Bucket.Min-1,
+				bucket.Shapes.S3Bucket.Max+1,
+			),
+			"desc": fmt.Sprintf(
+				"Bucket names must be between %d (min) and %d (max) characters long.",
+				bucket.Shapes.S3Bucket.Min,
+				bucket.Shapes.S3Bucket.Max,
+			),
+		},
+		{
+			"regx": "[^a-z0-9\\.\\-]",
+			"desc":  "Bucket names can consist only of lowercase letters, numbers, dots (.), and hyphens (-).",
+		},
+		{
+			"regx": "^[^a-z0-9]",
+			"desc":  "Bucket names must begin with a lowercase letter or number.",
+		},
+		{
+			"regx": "[^a-z0-9]$",
+			"desc":  "Bucket names must end with a lowercase letter or number.",
+		},
+		{
+			"regx": "\\.\\.",
+			"desc":  "Bucket names must not contain two adjacent periods.",
+		},
+		{
+			"regx": "^\\b((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\\.|$)){4}\\b$",
+			"desc":  "Bucket names must not be formatted as an IP address (for example, 192.168.5.4).",
+		},
+		{
+			"regx": "^(xn--|sthree-dwa|sthree-configurator21)",
```

Comment:
Avoid merging these together as to keep the descriptions from the official AWS docs. Same for suffixes.

Reply:
OK, so:
```go		
{
  Regexp:      *regexp.MustCompile("^xn--"),
  Description: "Bucket names must not start with the prefix 'xn--'.",
},
{
  Regexp:      *regexp.MustCompile("^(sthree-dwa|sthree-configurator21)"),
  Description: "Bucket names must not start with the prefix 'sthree-' and the prefix 'sthree-configurator'.",
},
{
  Regexp:      *regexp.MustCompile("-s3alias$"),
  Description: "Bucket names must not end with the suffix '-s3alias'.",
},
{
  Regexp:      *regexp.MustCompile("--ol-s3$"),
  Description: "Bucket names must not end with the suffix '--ol-s3'.",
},

```

Reply:
Yep!